File: frontend/src/lib/types/admin-health.ts
// Mirrors backend/app/schemas/admin_health.py exactly — keep these two in sync.

export interface DatabaseHealth {
  healthy: boolean;
  pool_checked_out: number | null;
  pool_size: number | null;
  error: string | null;
}

export interface RedisHealth {
  healthy: boolean;
  used_memory_bytes: number | null;
  error: string | null;
}

export interface StorageBackendHealth {
  name: string;
  healthy: boolean;
  error: string | null;
}

export interface CeleryQueueHealth {
  queue: string;
  healthy: boolean;
  worker_count: number;
}

export interface CeleryHealth {
  healthy: boolean;
  queues: CeleryQueueHealth[];
  error: string | null;
}

export interface DiskHealth {
  total_bytes: number;
  used_bytes: number;
  free_bytes: number;
  percent_used: number;
}

export interface LastBackupSummary {
  id: string;
  status: string;
  started_at: string;
  completed_at: string | null;
}

export interface DetailedHealthResponse {
  checked_at: string;
  database: DatabaseHealth;
  redis: RedisHealth;
  storage: StorageBackendHealth[];
  celery: CeleryHealth;
  disk: DiskHealth;
  last_backup: LastBackupSummary | null;
}

File: backend/app/schemas/admin_health.py
from datetime import datetime

from pydantic import BaseModel


class DatabaseHealth(BaseModel):
    healthy: bool
    pool_checked_out: int | None
    pool_size: int | None
    error: str | None = None


class RedisHealth(BaseModel):
    healthy: bool
    used_memory_bytes: int | None
    error: str | None = None


class StorageBackendHealth(BaseModel):
    name: str
    healthy: bool
    error: str | None = None


class CeleryQueueHealth(BaseModel):
    queue: str
    healthy: bool
    worker_count: int


class CeleryHealth(BaseModel):
    healthy: bool
    queues: list[CeleryQueueHealth]
    error: str | None = None


class DiskHealth(BaseModel):
    total_bytes: int
    used_bytes: int
    free_bytes: int
    percent_used: float


class LastBackupSummary(BaseModel):
    id: str
    status: str
    started_at: datetime
    completed_at: datetime | None


class DetailedHealthResponse(BaseModel):
    checked_at: datetime
    database: DatabaseHealth
    redis: RedisHealth
    storage: list[StorageBackendHealth]
    celery: CeleryHealth
    disk: DiskHealth
    last_backup: LastBackupSummary | None
